anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/source/data/initializer/DataInitializer.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2018-2019 Expedia Group, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expedia.adaptivealerting.anomdetect.source.data.initializer;
+
+import com.expedia.adaptivealerting.anomdetect.detect.Detector;
+import com.expedia.adaptivealerting.anomdetect.detect.MappedMetricData;
+import com.expedia.adaptivealerting.anomdetect.detect.outlier.algo.forecasting.ForecastingDetector;
+import com.expedia.adaptivealerting.anomdetect.forecast.point.algo.seasonalnaive.SeasonalNaivePointForecaster;
+import com.expedia.adaptivealerting.anomdetect.source.data.DataSource;
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+
+@AllArgsConstructor
+@Slf4j
+public class DataInitializer {
+    //TODO Find appropriate earliest time and max data point values.
+    private static final String EARLIEST_TIME = "7d";
+    private static final Integer MAX_DATA_POINTS = 2016;
+
+    @NonNull
+    private DataSource dataSource;
+
+    public void initializeDetector(MappedMetricData metricData, Detector detector) {
+        if (detector != null && "seasonalnaive".equals(detector.getName())) {
+            val forecastingDetector = (ForecastingDetector) detector;
+            val seasonalNaivePointForecaster = (SeasonalNaivePointForecaster) forecastingDetector.getPointForecaster();
+            val data = getHistoricalData(metricData);
+            seasonalNaivePointForecaster.getBuffer().initBuffer(data);
+        }
+    }
+
+    private double[] getHistoricalData(MappedMetricData mappedMetricData) {
```

Comment:
We should preserve the timestamps of the data.  Some timestamps will be missing.  The SeasonalBuffer knows how to fill in the gaps but it needs the timestamps.
Returning a list of `DataSourceResult`s is preferable.

See suggested code above.